docs/FCM_IMPLEMENTATION.md

```diff
@@ -0,0 +1,372 @@
+# FCM Implementation for MostroP2P Mobile
+
+## Overview
+
+This document outlines the implementation of Firebase Cloud Messaging (FCM) for push notifications in the MostroP2P mobile application. The implementation follows a privacy-preserving approach similar to **[MIP-05 (Marmot Push Notifications)](https://github.com/marmot-protocol/marmot/pull/18)**, where:
+
+- **Silent push notifications** wake up the app without exposing message content
+- **All message decryption happens locally** on the device
+- **Minimal metadata** is exposed to Firebase/Google services
+- **Custom notification server** ([mostro-push-server](https://github.com/AndreaDiazCorreia/mostro-push-server)) handles token management and notification delivery
```

Comment:
great job @AndreaDiazCorreia , please move mostro-push-server repository to MostroP2P github organization and fix this reference